identity/src/androidTest/java/com/android/identity/IdentityCredentialTest.java

```diff
@@ -0,0 +1,172 @@
+import junit.framework.AssertionFailedError;
+import org.junit.After;
+import org.junit.Assume;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+
+/**
+ * Tests {@link IdentityCredential}.
+ */
+@RunWith(Parameterized.class)
+public class IdentityCredentialTest {
+
+  private static final String CREDENTIAL_NAME = "fake credential for test";
+  private static final int CIPHER_SUITE = CIPHERSUITE_ECDHE_HKDF_ECDSA_WITH_AES_256_GCM_SHA256;
+
+  private IdentityCredentialStore store;
+
+  public IdentityCredentialTest(Object store, Object storeName) {
+    this.store = Preconditions.checkNotNull((IdentityCredentialStore) store);
+    Preconditions.checkNotNull((String) storeName); // only used for test name
+  }
+
+  /**
+   * Tests storing static authentication data of varying size for varying numbers of auth keys.
+   */
+  @MediumTest
+  @Test
+  public void storeStaticAuthenticationData()
+          throws IdentityCredentialException {
+    checkStaticAuthData(1, 100); // 1 key x 100 bytes of static authentication data = 100 Bytes
+    checkStaticAuthData(3, 100);
+    checkStaticAuthData(100, 100); // 100 keys x 100 Bytes = 10 KBytes
+    checkStaticAuthData(1, 100000); // 1 key x 100 KBytes
+  }
+
+  @LargeTest
+  @Test
+  public void storeStaticAuthenticationDataLarge()
+          throws IdentityCredentialException {
+    Assume.assumeTrue("Test fails on API <= 28", VERSION.SDK_INT >= 29 /* Android 10+ */);
+    // Per b/234563696, I've confirmed these failing on:
+    // Pixel 2 API 24
+    // Pixel 2 API 26
+    // Pixel 3 API 28
+    checkStaticAuthData(10, 10000); // fails
+    checkStaticAuthData(100, 1000); // flaky
+    checkStaticAuthData(100, 10000); // fails
+  }
+
+  private void checkStaticAuthData(int numAuthKeys, int staticAuthDataSizeBytes)
+          throws IdentityCredentialException {
+    final int usesPerKey = 3;
+    ProvisioningTest.createCredential(store, CREDENTIAL_NAME);
+    try {
+      IdentityCredential credential = Preconditions.checkNotNull(getCredentialIfPresent());
+
+      storeFakeStaticAuthData(credential, usesPerKey, numAuthKeys, staticAuthDataSizeBytes);
+
+      // Check that this doesn't throw (it did throw in internal b/234563696).
+      try {
+        getCredentialIfPresent();
+      } catch (RuntimeException e) {
+        String msg = String.format(Locale.US,
+                "Failed to load credential for %d authKeys with %d bytes static auth data each",
+                numAuthKeys, staticAuthDataSizeBytes);
+        throw new IdentityCredentialException(msg, e);
+      }
+    } finally {
+      deleteCredentialFromStoreIfPresent();
+    }
+  }
+
+  @Before
+  public void setUp() {
+    deleteCredentialFromStoreIfPresent(); // clean up in case a previous test didn't
+  }
+
+  @After
+  public void tearDown() {
+    deleteCredentialFromStoreIfPresent();
+    store = null;
```

Comment:
This seems odd. Is this typical of java tests to explicitly null things on `tearDown`?

This means that the object can never be used again. Is that valid? Shouldn't we be able to call `setUp` on the same object and run the tests again (pedantically speaking).

Reply:
Removed.

The reason I added it is that the JUnitRunner (and AndroidJUnitRunner) has the annoying (IMO misdesigned property) that it keeps instances of the *Test classes referenced until the very end of the test run. This means that all values referenced by *Test class instances would continue to occupy heap memory until the overall test run finishes. This has led to lots of death-by-papercut instances in CTS (Compatibility Test Suite) because the large number of Test class instances run prevent a lot of memory from being garbage collected.

However, the bug here really should be fixed in the runner (perhaps it already has), plus these particular tests are unlikely to run as part of a very large suite. Also, because it's very nonobvious that store is actually initialized in the constructor, because the constructor is parameterized (which makes it similar to @Before / setUp()), this is confusing to readers and therefore doesn't pull its weight.